Add steps for cleaning task data to Software (#2263)
astronomer/docs: software/configure-deployment.md

## Programmatically create or update Deployments

You can programmatically create or update Deployments with all possible configurations using the Houston API `upsertDeployment` mutation. See [Create or update a Deployment with configurations](houston-api.md#create-or-update-a-deployment-with-configurations).

## Clean Deployment task metadata

You can run a cron job to automatically archive task and DAG metadata from your Deployment. This job runs [`airflow db clean`](https://airflow.apache.org/docs/apache-airflow/stable/cli-and-env-variables-ref.html#clean) for all of your Deployments and exports the results for each Deployment as a file to your external storage service. To run this job for a Deployment, you must install the Astronomer-maintained `airflow-dbcleanup-plugin` on the Deployment. 

> 1. For each of your Deployments, add the following line to the `Dockerfile` of your Deployment's Astro project. Replace `<latest-version>` with the latest available version on the [`airflow-dbcleanup-plugin` GitHub repository](https://github.com/astronomer/airflow-dbcleanup-plugin/releases).

Author: ```suggestion
1. For each of your Deployments, add the following line to the `requirements.txt` file of your Deployment's Astro project. Replace `<latest-version>` with the latest available version in the [`airflow-dbcleanup-plugin` GitHub repository](https://github.com/astronomer/airflow-dbcleanup-plugin/releases).
```

    ```text

>     RUN pip install https://github.com/astronomer/airflow-dbcleanup-plugin/releases/download/<latest-version>/astronomer_dbcleanup_plugin-<latest-version>-py3-none-any.whl

Author: ```suggestion
    https://github.com/astronomer/airflow-dbcleanup-plugin/releases/download/<latest-version>/astronomer_dbcleanup_plugin-<latest-version>-py3-none-any.whl
```

    ```

> 2. Configure an Airflow connection to your external storage service in JSON or URI format so that it can be stored as an environment variable. See [Airflow documentation](https://airflow.apache.org/docs/apache-airflow/stable/howto/connection.html#storing-connections-in-environment-variables).

Author: ```suggestion
2. Configure an Airflow connection to your external storage service in JSON or URI format so that it can be stored as an environment variable. You must use a service account to authenticate to your service. See [Airflow documentation](https://airflow.apache.org/docs/apache-airflow/stable/howto/connection.html#storing-connections-in-environment-variables) to learn how to configure your connection.
```

> 3. Store the connection information as a Kubernetes Secret on your Astronomer cluster. See [Kubernetes documentation](https://kubernetes.io/docs/concepts/configuration/secret/#creating-a-secret).

Author: ```suggestion
3. Store the connection environment variable as a Kubernetes Secret on your Astronomer cluster. See [Kubernetes documentation](https://kubernetes.io/docs/concepts/configuration/secret/#creating-a-secret).
```

> 4. Add the following configuration to your `config.yaml` file and the default values as needed.

Author: ```suggestion
4. Add the following configuration to your `config.yaml` file and change the default values as needed.
```

   
    ```yaml
    houston:
      cleanupAirflowDb:
        # Enable cleanup CronJob
        enabled: false
     
        # Default run is at 5:23 every morning https://crontab.guru/#23_5_*_*_*
        schedule: "23 5 * * *"
     
        # Cleanup deployments older than this many days
        olderThan: 365
     
        # Output path of archived data csv export
        outputPath: "/tmp"
     
        # Delete archived tables
        purgeArchive: true
     
        # Print out the deployments that should be cleaned up and skip actual cleanup
        dryRun: false
     
        # Name of file storage provider, supported providers - aws/azure/gcp/local
        provider: local
     
        # Name of the provider bucket name / local file path

>         bucketName: "/tmp"

Reviewer: @rishkarajgi we are by default prefixing with /tmp for local provider, we can just give a folder name

     
        # The name of the Kubernetes Secret containing your Airflow connection

>         providerEnvSecretName: ""

Author: ```suggestion
        providerEnvSecretName: "<your-secret-name>"
```
